### ISSUE TITLE
MJPEGStream throws InvalidOperationException in .NET Core 2.0

### ISSUE THREAD

**Issue by stigvoss** [accord-net/framework#700]
Related to adding support for .NET Core #318 and MJPEGStream issue #699 
MJPEGStream class throws an InvalidOperationException when the WorkerThread attempts to set ReadTimeout on response stream containing the message:

> Timeouts are not supported on this stream.

This only concerns .NET Core 2.0 and works as expected in full .NET Framework.

The source of the exception is [here](https://github.com/accord-net/framework/blob/3.6.0/Sources/Accord.Video/MJPEGStream.cs#L531), the property is seemingly not supported in .NET Core 2.0.

The following fails with described exception:
```
MJPEGStream stream = new MJPEGStream(source)
{
    Login = ...,
    Password = ...,
    SeparateConnectionGroup = false
};
stream.NewFrame += OnFrameReceived;
stream.VideoSourceError += OnSourceError;
stream.Start();
```

**Comment by stigvoss** [comment]
I seem to observe that .NET Framework supplies a ConnectStream where .NET Core supplies a WinHttpResponseStream.

**Comment by cesarsouza** [comment]
Hi @stigvoss,

Thanks a lot for reporting the issue! Do you think you could contribute with a failing unit test triggering the problem, so it could be easier to fix?

Thanks!

Also, if you (or anyone reading this in the future) would happen to know what would have to be done in this case, please do not hesitate to comment and/or send a pull request. It would take a while until I could have enough time to dig deep into the issue.

Regards,
Cesar

**Comment by stigvoss** [comment]
I am not entirely sure how to achieve the successful creation of a unit test in this scenario without performing significant refactoring to the code as the exception surfaces within a worker thread and the only external signs of the issue is apparent through a callback on the VideoSourceError event.
At the same time, the code requires an external HTTP source for MJPEG to trigger the exception and I assume it would require a hard-coded URL as a source. I do not see this as a viable choice for unit testing.

Without knowing all the potential consequences, I would suggest the same type of solution as proposed in #699, at least as a temporary solution. The drawback found during some manual testing, is that the stream will never trigger the VideoSourceError to surface the error and it shows that the lack of timeout prevents the stream from recovering after a failed connection.

**Comment by stigvoss** [comment]
I believe I may have found a solution or work around for .NET Standard 2.0 based on a cancellation token with a timeout and ReadAsync on the stream.

The solution may be looking somewhat similar to this:
```
TimeSpan timeout = TimeSpan.FromMilliseconds(requestTimeout);
CancellationTokenSource tokenSource = new CancellationTokenSource(timeout);
Task<int> task = stream.ReadAsync(buffer, total, readSize, tokenSource.Token);

task.Wait();

if ((read = task.Result) == 0)
    throw new ApplicationException();
```

Instead of the current
```
if ((read = stream.Read(buffer, total, readSize)) == 0)
    throw new ApplicationException();
```

I will create a fix, test it and drop a pull request if it works as expected.

**Comment by cesarsouza** [comment]
Thanks a lot for the investigation and the proposed fix!

**Comment by cesarsouza** [comment]
Available in 3.7.0.
